base/experimental.jl

```diff
@@ -275,17 +275,22 @@ macro overlay(mt, def)
     esc(def)
 end
 
+let new_mt(name::Symbol, mod::Module) = begin
+        ccall(:jl_check_top_level_effect, Cvoid, (Any, Cstring), mod, name)
```

Comment:
Why 2 ccalls? That seems like double the chance for problems.

Reply:
The `name` parameter in this version seems wrong too, since it is expecting the name of the caller function (`@MethodTable`) not the value of the argument to that function.